ganalytics.php

```diff
@@ -117,25 +107,25 @@ function gaOptout() {
      * Return all personalized GA settings
      * @param string $trackingId Google Analytics Tracking ID
      * @param string $objectName Global variable name for the GA object
-     * @return array
+     * @return string
      */
     private function getTrackingSettings($trackingId, $objectName)
     {
-        $cookie_config = $this->getCookieConfiguration();
-
-        $settings = [
-          'trace-debug' =>  "window.ga_debug = {trace: true};",
-          'create'      => "{$objectName}('create', '{$trackingId}', {$cookie_config});",
-          'anonymize'   => "{$objectName}('set', 'anonymizeIp', true);",
-          'force-ssl'   => "{$objectName}('set', 'forceSSL', true);",
-          'send'        => "{$objectName}('send', 'pageview');"
-        ];
+        if ($this->config->get('plugins.ganalytics.debugMode', false)) {
+            $settings['debug_mode'] = true;
+        } else {
+            $settings = [];
+        }
 
-        if (!$this->config->get('plugins.ganalytics.debugTrace', false)) unset ($settings['trace-debug']);
-        if (!$this->config->get('plugins.ganalytics.anonymizeIp', false)) unset ($settings['anonymize']);
-        if (!$this->config->get('plugins.ganalytics.forceSsl', false)) unset ($settings['force-ssl']);
+        $settings = array_merge($settings, $this->getCookieConfiguration());
+        if (!empty($settings)) {
+            $settings = str_replace('"', '\'', json_encode($settings));
```

Comment:
**Question:** Why is this replacement needed? Doesn't Google handle the default JSON escaping?

Reply:
It was more a cosmetic fix really, but when testing various invalid JSON strings I have found that it messes with validity of the final output. Removed now.